### PR TITLE
test: port multinode test to Go

### PR DESCRIPTION
**Pull request by guseggert**
The sharness version of this test takes ~130 seconds to run, this one takes 0.8 seconds.

This doesn't fully port the test 1:1 because this test isn't actually doing what it looks like. It looks like it's testing the Websocket transport with "listentype,ws" IPTB attribute, but that attribute doesn't exist in ipfs/iptb-plugins, so it does nothing, so the test actually just runs the same test twice (Yamux disabled). So I've not ported the no-op Websocket 
portion of the test.

This also moves equivalent functionality from jbenet/go-random-files into the testutils package. This just copies the code and modifies it slightly for better ergonomics.

### CODE REVIEW

File: test/cli/multinode_test.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	"github.com/stretchr/testify/require"
+)
+
+func TestMultipleNodes(t *testing.T) {
+	t.Parallel()
+
+	checkSingleFile := func(nodes harness.Nodes) {
+		s := testutils.RandomStr(100)
+		hash := nodes[0].IPFSAddStr(s)
+		nodes.ForEachPar(func(n *harness.Node) {
```
Comment (Jorropo) [inline test/cli/multinode_test.go:21]: `Par` ? Quèsaco ?
Should it be `ForEachPeer` ?

File: test/cli/multinode_test.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	}
+
+	t.Run("vanilla nodes", func(t *testing.T) {
+		t.Parallel()
+		nodes := harness.NewT(t).NewNodes(5).Init().StartDaemons().Connect()
+		checkSingleFile(nodes)
+		checkRandomDir(nodes)
+	})
+
+	t.Run("nodes with yamux disabled", func(t *testing.T) {
```
Comment (Jorropo) [inline test/cli/multinode_test.go:48]: I think this should be a table based that tests all muxers and all transports.

If you think this is expending the scope too much feel free to resolve this I'll open a new PR.

File: test/cli/multinode_test.go (hunk 1 above)
Comment (guseggert) [reply test/cli/multinode_test.go:21]: nope it's "par" for "parallel", it invokes those funcs in parallel w/ goroutines

File: test/cli/multinode_test.go (hunk 2 above)
Comment (guseggert) [reply test/cli/multinode_test.go:48]: There are some similar tests in `t0125-twonode.sh` testing TCP, mplex, yamux, noise, TLS, QUIC, and WebTransport. I would just port that over.  I honestly think this "multinode" test isn't really adding much value, I'd be okay with just deleting it entirely too.

**Comment by guseggert** [comment]
closing in favor of https://github.com/ipfs/kubo/pull/9809

**Pull request closed without merging**